### PR TITLE
Remove /api/applications/{draft,save}

### PR DESCRIPTION
**Pull request by luontola**
#852

### CODE REVIEW

File: src/clj/rems/api/applications.clj [hunk 1]
```diff
@@ -180,18 +149,22 @@
...
-    (GET "/attachments/" []
+    (GET "/attachments" []
       :summary "Get an attachment for a field in an application"
       :roles #{:logged-in}
       :query-params [application-id :- (describe s/Int "application id")
                      field-id :- (describe s/Int "application form field id the attachment is related to")]
-      (let [form (applications/get-form-for (getx-user-id) application-id)]
+      (let [user-id (getx-user-id)
+            application (->> (applications/get-dynamic-application-state application-id)
+                             (dynamic/assoc-possible-commands user-id))]
```
Comment (opqdonut) [inline src/clj/rems/api/applications.clj]: should this combination of functions have a name?

File: src/clj/rems/db/form.clj [hunk 2]
```diff
@@ -63,125 +57,3 @@
...
-(defn- check-for-disabled-items! [items]
-  (let [disabled-items (remove :enabled items)]
-    (when (seq disabled-items)
-      (throw (rems.InvalidRequestException. (str "Disabled catalogue items " (pr-str disabled-items)))))))
-
-(defn- create-new-draft-for-items [user-id catalogue-item-ids]
-  (let [draft (applications/make-draft-application user-id catalogue-item-ids)]
-    (check-for-disabled-items! (getx draft :catalogue-items))
-    (let [wfid (getx draft :wfid)
-          id (applications/create-new-draft user-id wfid)]
```
Comment (opqdonut) [inline src/clj/rems/db/form.clj]: can the cyclic dependency applications->form->applications be broken now?

File: test/clj/rems/api/test_applications.clj [hunk 3]
```diff
@@ -256,102 +292,36 @@
...
-                 app))))
-      (testing "saving fields as other user is forbidden"
-        (is (response-is-forbidden?
-             (-> (request :post "/api/applications/save")
-                 (authenticate api-key "bob")
-                 (json-body {:command "save"
-                             :application-id application-id
-                             :items {1 "xxxxx"}})
-                 app))))
-      (testing "can't submit with missing required fields"
```
Comment (opqdonut) [inline test/clj/rems/api/test_applications.clj]: where did these validation tests go?

Comment (luontola) [reply test/clj/rems/api/test_applications.clj]: It's covered by https://github.com/CSCfi/rems/blob/dead-code/src/clj/rems/workflow/dynamic.clj#L938-L960 and the test for maxlength was made into a unit test at https://github.com/CSCfi/rems/blob/dead-code/test/clj/rems/test_form_validation.clj#L49-L59

**Pull request merged**